Add static credential support to RabbitMQ secret engine (#21644)
hashicorp/vault: website/content/docs/secrets/rabbitmq.mdx

    more about [RabbitMQ management tags][rmq-perms]
    and [RabbitMQ topic authorization][rmq-topics].

> Similarly, if a static username is required, one can configure a static role
> instead:
> 
> 1.  Configure a role that maps a name in Vault to virtual host permissions:

Reviewer: Suggest folding the new instructions into the existing step as an alternative rather than adding a second list with only one step.

```suggestion
  Alternatively, if you require a static username, configure a static role
  that maps a name in Vault to virtual host permissions:
```

style correction: active voice, speak directly to the reader with "you",

    ```text
    $ vault write rabbitmq/static-roles/my-static-role \

>         username="foobar" \

Reviewer: ```suggestion
        username="vhost" \
```
style correction: avoid placeholders like "foo", "bar", and "foobar"

        rotation_period=300 \
        vhosts='{"/":{"write": ".*", "read": ".*"}}'
    Success! Data written to: rabbitmq/static-roles/my-static-role
    ```

>     By writing to the `static-roles/my-static-role` path we are defining the
>     `my-static-role` static role. The way this role is created is similar to
>     the role in the example above. However, it will always keep `foobar` as a
>     username, and automatically rotate the password every 300 seconds. Hence
>     it will not create a new password every time credentials are requested.

Reviewer: ```suggestion
Writing to the `static-roles/my-static-role` defines a static role called
`my-static-role`. The creation process is similar to the previous example,
but the role keeps `vhost` as a username and automatically rotate the
password every 300 seconds rather than creating a new password every time
Vault receives a credential request.
```

style correction: active voice

    such that trusted operators can manage the role definitions, and both users
    and applications are restricted in the credentials they are allowed to read.

> 1.  Alternatively, request credentials for a static role by reading from the

Reviewer: ```suggestion
Alternatively, request credentials for a static role by reading from the
```

Missed this one in the first review, apologies. No need to number steps if there's only one step :)

    such that trusted operators can manage the role definitions, and both users
    and applications are restricted in the credentials they are allowed to read.

1.  Alternatively, request credentials for a static role by reading from the

>     `/static-creds` endpoint with hte name of the static role:

Reviewer: ```suggestion
    `/static-creds` endpoint with the name of the static role:
```

correct typo

    of the role:

    ```text
    $ vault read rabbitmq/static-creds/my-static-role
    Key                    Value
    ---                    -----
    last_vault_rotation    2023-07-10T13:52:51.007373395Z
    password               5KZYIzbodxzyE5t1CzpPNeTe9t4gz9Wdx023
    rotation_period        5m
    ttl                    4m54s

>     username               foobar

Reviewer: ```suggestion
    username               vhost
```

Change username to reflect suggestion in previous code block
